File: c++/브루트포스/일곱난쟁이.cpp
#include <iostream>

using namespace std;

int main(void){
    cin.tie(NULL);
    ios::sync_with_stdio(false);

    int arr[9+1] = {0 , };
    int sum=0;
    int ret[7+1] = {0 , };
    for(int i=0;i<9;i++){
        cin>>arr[i];
        sum += arr[i];
    }
    //총합 sum(9개의 합)에서 2개씩 더한 값을 빼서 100인걸 찾는다.
    int cnt=0;
    for(int i=0;i<9;i++){
        for(int j=i+1;j<9;j++){
            int tmp = arr[i] + arr[j];
            if(sum-tmp == 100){
                for(int k=0;k<9;k++){
                    if(k==i || k==j) continue;
                    ret[cnt++] = arr[k];
                }
            }
        }
    }

    for(int i=0;i<7;i++){
        for(int j=i+1;j<7;j++){
            if(ret[j] < ret[i]){
                int tmp=ret[j];
                ret[j]=ret[i];
                ret[i]=tmp;
            }
        }
    }

    for(int i=0;i<7;i++){
        cout<<ret[i]<<" ";
    }

    
    return 0;
}

File: python/문제집/라이 문제집/일곱난쟁이.py
import sys

input = lambda: sys.stdin.readline().strip()

arr = []
ans = [0] * 10
sums = 0
for _ in range(9):
    n = int(input())
    arr.append(n)
    sums += n

arr.sort()
cnt = 0
# 총합 sum(9개의 합)에서 2개씩 더한 값을 빼서 100인걸 찾는다.
for i in range(0, 9, 1):
    for j in range(i + 1, 9, 1):
        tmp = arr[i] + arr[j]
        if sums - tmp == 100:
            for k in range(0, 9, 1):
                if k == i or k == j:
                    continue
                ans[cnt] = arr[k]
                cnt += 1

for i in ans:
    if i == 0:
        break
    print(i)
